internal-dns/src/names.rs

```diff
@@ -84,4 +84,9 @@ impl ServiceName {
             }
         }
     }
+
+    /// Returns the full DNS name of this service
+    pub fn srv_name(&self) -> String {
```

Comment:
```suggestion
    pub(crate) fn srv_name(&self) -> String {
```

suggesting this just to make it harder for people to accidentally use the DNS name for something.

Reply:
I was planning on letting callers construct progenitor clients using this name as part of the URL during construction -- would that be wrong?

For example, see: `sled-agent/src/nexus.rs` in this PR -- I use this SRV name when making the progenitor client.